// @biyy/format-rupiah-datetime/index.js
/**
 * Mengubah angka menjadi format mata uang Rupiah (IDR).
 * Contoh: 150000 menjadi "Rp 150.000"
 * @param {number} number - Angka yang akan diformat.
 * @returns {string} String dalam format Rupiah.
 */
function formatRupiah(number) {
  if (isNaN(number)) {
    return 'Invalid Number';
  }
  return new Intl.NumberFormat('id-ID', {
    style: 'currency',
    currency: 'IDR',
    minimumFractionDigits: 0
  }).format(number);
}

/**
 * Mengubah string tanggal (date string) menjadi format lokal Indonesia.
 * Contoh: "2023-11-12T10:00:00.000Z" menjadi "12 November 2023"
 * @param {string} dateString - String tanggal yang valid.
 * @returns {string} String tanggal dalam format Indonesia.
 */
function formatDate(dateString) {
  if (!dateString) return '';
  try {
    return new Date(dateString).toLocaleDateString('id-ID', {
      day: '2-digit',
      month: 'long',
      year: 'numeric'
    });
  } catch (error) {
    return 'Invalid Date';
  }
}

/**
 * Mengubah string tanggal (date string) menjadi format tanggal dan waktu.
 * Contoh: "2023-11-12T10:30:00.000Z" menjadi "12 November 2023 17.30"
 * @param {string} dateString - String tanggal yang valid.
 * @returns {string} String tanggal dan waktu dalam format Indonesia.
 */
function formatDateTime(dateString) {
  if (!dateString) return '';
  try {
    return new Date(dateString).toLocaleString('id-ID', {
      day: '2-digit',
      month: 'long',
      year: 'numeric',
      hour: '2-digit',
      minute: '2-digit'
    });
  } catch (error) {
    return 'Invalid Date';
  }
}

// Ekspor semua fungsi agar bisa digunakan di proyek lain
module.exports = {
  formatRupiah,
  formatDate,
  formatDateTime
};
